### PR TITLE
Fixed up support for list content types when used with Gemini

### PR DESCRIPTION
**Pull request by ivanleomk**
This allows us to now support calls like this below

```
import instructor
import google.generativeai as genai
from pydantic import BaseModel
from pydub import AudioSegment


client = instructor.from_gemini(
    client=genai.GenerativeModel(
        model_name="models/gemini-1.5-flash-latest",
    ),
    mode=instructor.Mode.GEMINI_JSON,
)


sound = AudioSegment.from_mp3("sample.mp3")

content = [
    "Please transcribe this recording:",
    {"mime_type": "audio/mp3", "data": sound[:10000].export().read()},
]


class Description(BaseModel):
    description: str


resp = client.create(
    response_model=Description,
    messages=[{"role": "user", "content": content}],
)

print(resp)
```

Which we couldn't have supported before.

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit b87f60fa388c26396921ba2d9527be49a4f741a8  | 
|--------|--------|

feat: add support for list content types in Gemini

### Summary:
This PR adds support for list content types in Gemini, updates utility functions, and includes tests, documentation, and a code example for handling audio data.

**Key points**:
- Add support for list content types in Gemini.
- Update `transform_to_gemini_prompt()` for list handling.
- Introduce `get_message_content()` in `instructor/utils.py`.
- Add tests: `test_list_content.py`, `test_multimodal_content.py`, `test_roles.py`.
- Include `sample.mp3` for testing.
- Update documentation: `multi_modal_gemini.md`, `index.md`, `mkdocs.yml`.
- Provide a code example demonstrating new functionality.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Comment by ivanleomk** [comment]
CI tests

```
============================== test session starts ===============================
platform darwin -- Python 3.11.9, pytest-7.4.4, pluggy-1.5.0
rootdir: /Users/ivanleo/Documents/coding/instructor
plugins: examples-0.0.10, anyio-4.4.0, asyncio-0.21.2
asyncio: mode=Mode.STRICT
collected 110 items                                                              

tests/llm/test_gemini/test_list_content.py .                               [  0%]
tests/llm/test_gemini/test_modes.py ........                               [  8%]
tests/llm/test_gemini/test_multimodal_content.py ..                        [ 10%]
tests/llm/test_gemini/test_patch.py ........                               [ 17%]
tests/llm/test_gemini/test_retries.py ........                             [ 24%]
tests/llm/test_gemini/test_simple_types.py ...                             [ 27%]
tests/llm/test_gemini/test_stream.py ............                          [ 38%]
tests/llm/test_gemini/evals/test_classification_enums.py ................. [ 53%]
...                                                                        [ 56%]
tests/llm/test_gemini/evals/test_classification_literals.py .............. [ 69%]
......                                                                     [ 74%]
tests/llm/test_gemini/evals/test_entities.py ....                          [ 78%]
tests/llm/test_gemini/evals/test_extract_users.py ............             [ 89%]
tests/llm/test_gemini/evals/test_sentiment_analysis.py ............        [100%]

================================ warnings summary ================================
```

**Comment by dukeeagle** [comment]
Beautiful!

**Review by jxnl — APPROVED** [review]

File: docs/index.md [hunk 1]
```diff
@@ -214,9 +218,181 @@ assert resp.name == "Jason"
...
+
+    mp3_file = genai.upload_file("./sample.mp3") #(2)!
+
+
+    class Description(BaseModel):
+        description: str
+
 
-**Note**: Gemini Tool Calling is still in preview, and there are some limitations. You can learn more about them in the [Vertex AI examples notebook](../hub/vertexai.md).
+    resp = client.create(
```
Comment (jxnl) [inline docs/index.md:249]: this is way too long lets move this to a gemini setion

**Review by jxnl — COMMENTED** [review]
move gemini into its own page so its easier to find 

structured output with multi modal gemini or something

**Comment by ivanleomk** [comment]
Added an additional test to check for support for parsing of assistant and system messages. Previously we would get errors if we did something like this

```
import instructor
import google.generativeai as genai
from pydantic import BaseModel
from pydub import AudioSegment


client = instructor.from_gemini(
    client=genai.GenerativeModel(
        model_name="models/gemini-1.5-flash-latest",
    ),
    mode=instructor.Mode.GEMINI_JSON,
)


class Description(BaseModel):
    description: str


resp = client.create(
    response_model=Description,
    messages=[
        {
            "role": "system",
            "content": "You are a helpful assistant that describes what a sunset in the desert looks like.",
        },
    ],
)

print(resp)
```

We would get the following errors

```
Traceback (most recent call last):
  File "/Users/ivanleo/Documents/coding/instructor/./main.py", line 19, in <module>
    resp = client.create(
           ^^^^^^^^^^^^^^
  File "/Users/ivanleo/Documents/coding/instructor/instructor/client.py", line 116, in create
    return self.create_fn(
           ^^^^^^^^^^^^^^^
  File "/Users/ivanleo/Documents/coding/instructor/instructor/patch.py", line 140, in new_create_sync
    response_model, new_kwargs = handle_response_model(
                                 ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ivanleo/Documents/coding/instructor/instructor/process_response.py", line 488, in handle_response_model
    new_kwargs = update_gemini_kwargs(new_kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ivanleo/Documents/coding/instructor/instructor/utils.py", line 335, in update_gemini_kwargs
    kwargs["contents"] = transform_to_gemini_prompt(kwargs.pop("messages"))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ivanleo/Documents/coding/instructor/instructor/utils.py", line 272, in transform_to_gemini_prompt
    messages_gemini[0]["parts"].insert(0, f"*{system_prompt}*")
    ~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```

All 111 tests are passing with the new change too

```
tests/llm/test_gemini/test_list_content.py .                                                                       [  0%]
tests/llm/test_gemini/test_modes.py ........                                                                       [  8%]
tests/llm/test_gemini/test_multimodal_content.py ..                                                                [  9%]
tests/llm/test_gemini/test_patch.py ........                                                                       [ 17%]
tests/llm/test_gemini/test_retries.py ........                                                                     [ 24%]
tests/llm/test_gemini/test_roles.py .                                                                              [ 25%]
tests/llm/test_gemini/test_simple_types.py ...                                                                     [ 27%]
tests/llm/test_gemini/test_stream.py ............                                                                  [ 38%]
tests/llm/test_gemini/evals/test_classification_enums.py ....................                                      [ 56%]
tests/llm/test_gemini/evals/test_classification_literals.py ....................                                   [ 74%]
tests/llm/test_gemini/evals/test_entities.py ....                                                                  [ 78%]
tests/llm/test_gemini/evals/test_extract_users.py ............                                                     [ 89%]
tests/llm/test_gemini/evals/test_sentiment_analysis.py ............                                                [100%]
```

**Pull request merged**